src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -82,9 +85,21 @@ private ResponseEntity postEventInternal(final String eventTypeName,
             final JSONArray eventsAsJsonObjects = new JSONArray(eventsAsString);
 
             final int eventCount = eventsAsJsonObjects.length();
-            eventTypeMetrics.reportSizing(eventCount, eventsAsString.length());
+            final EventPublishResult result = publisher.publish(eventsAsJsonObjects, eventTypeName, client);
+            if (result.getStatus() == EventPublishingStatus.SUBMITTED) {
```

Comment:
I think it makes sense now to move this code to separate method (I mean the code that calculates metrics)